### PR TITLE
Use Redux in React components.

### PR DESCRIPTION
**Pull request by ZabihullahNooriWardak**
project requirements : 
Wrap your <App> with the <Provider> component from react-redux
Pass your configured store into the <Provider>
Display your books, received from the slice, in a reusable component
Dispatch actions using useDispatch
Add a <Button> component, which includes:
An event handler that adds a book to the books array (with attributes id, title and author)
Add a <Button> component, which includes:
An event handler that removes a book from the books array (by id)

### CODE REVIEW

File: src/components/addBookForm.js [hunk 1]
```diff
@@ -1,20 +1,44 @@
...
+  const dispatch = useDispatch();
+  const [bookName, setBookName] = useState('');
+  const [category, setCategory] = useState('');
+
+  const handleAddBook = (e) => {
+    e.preventDefault();
+    dispatch(addBook({ title: bookName, category }));
+    setBookName('');
+    setCategory('');
+  };
```
Comment (levy002) [inline src/components/addBookForm.js:15]: - [ ] It's crucial to maintain consistency and reliability in the way you manage books. To address an issue we've been encountering, you need to add a new property called `itemId` to each newly added book. This property is essential for removing books, and its absence has been causing unexpected behavior in your application. By including the `itemId` property for every new book, you ensure that your book management system works as expected, making it easier to identify and remove specific books when needed. This change will enhance the overall functionality and user experience of your app, providing a more seamless and error-free book management process.


```suggestion
  const handleAddBook = (e) => {
    e.preventDefault();
    dispatch(addBook({ title: bookName, category, itemId: Math.floor(Math.random() * 1000) }));
    setBookName('');
    setCategory('');
  };
```

**Review by levy002 — APPROVED** [review]
## Hello @ZabihullahNooriWardak   , :wave:
### I'm Levy, your PR reviewer

Good job so far :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No linter error :heavy_check_mark: 

## Status: REQUIRED CHANGES ♻️
_Check the comments under the review._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @levy002 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/bookList.js [hunk 2]
```diff
@@ -1,28 +1,37 @@
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
 import Book from './book';
 import '../style/booklist.css';
-import AddBook from './addBookForm';
+// eslint-disable-next-line import/extensions
```
Comment (V-Blaze) [inline src/components/bookList.js:5]: - [ ] As required by Microverse, you are not meant to use any linters diable rule on your codebase. The linters are there to make sure you follow the best practices. I'll suggest you remove this and try to fix the error. :smile: :small_airplane: :wave:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @ZabihullahNooriWardak 

# Almost there 🟡 

1094a3e620ebea4cd6cf2d0aceaa8f52


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing some changes from the previous reviewers, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
-  Correct Git fow✔
- Clean and organized code 💯 🥇 ✔

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- [OPTIONAL] When writing commit messages in GitHub, it's a good practice to start them with a capital letter. Clear and consistent commit messages help us maintain a well-organized repository. I'll suggest you take note of this when writing commit messages in your subsequent projects. :smile: :wave:  

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 ✔️ 

## Hi @ZabihullahNooriWardak 👋  

Great job implementing the required changes 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights 🎯 
✔️ Descriptive PR
✔️ Actions are dispatched via the UI 📌 
✔️ No linter errors

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**